tools/2025-03-workflow-changes.md

```diff
@@ -0,0 +1,260 @@
+
+All workflows can be run on branches so that fork owners do not have to merge the changes to the main branch of the fork. Having to run forks from the main branch makes fork synchronization more difficult.
+
+#### Workflows that perform PR checks should be configurable to run from the committing branch
+
+When a PR includes a change to a workflow, the workflow should be runnable from the branch if the person running the workflow has write access to the repo. (PRs submitted from forks where the contributor does not have write access should continue to require approval to run by someone with write access.)
+
+### Design options considered but not chosen
+
+* Adopt a new automation tool like [Just](https://github.com/casey/just) or [Task](https://taskfile.dev/), and deprecate Make. These tools have advantages, but a side-effect of adopting a new tool would likely be the existence of a new tool in our toolbox without the removal of Make, which would increase complexity. Make currently meets our needs for invoking developer automation.
+* Remove everything from GitHub workflows except the invocation of another tool: GitHub workflows have advantages when setting up the automation environment, e.g. retrieving credentials, cloning repos, configuring runners, installing tools, etc.
+
+## Security
+
+### Least Privilege
+
+Workflows will continue to adhere to the least privilege feature in which they declare which permissions are required by the workflow.
+
+### Identity
+
+Where possible, the automation logic will use the GitHub CLI, which has identity built in. Developers would log into the CLI when running code on their machines, which would allow them to run the logic on their forks. When running as part of at GitHub workflow, the GitHub CLI context is already set up and no further configuration is required.
+
+## Development plan
+
+### Docs: Contributing Automation Code
+
+A new folder exists in the `radius/docs` folder that provides a `README.md` file with guidance on creating and editing GitHub workflows and other repo automation like `make`.
+
+### PR Template
+
+Modify the PR template checklist for each repository that contains workflows to include items similar to the following:
+
+Workflow modifications have been tested on a repo fork and the changes conform to the workflow development guidance (add hyperlink to the guidance )  
+
+* [ ] This PR contains no modifications to GitHub workflows or repo automation.  
+* [ ] Changes to GitHub workflows, actions, and other repo automation have been tested on a repo fork.
+
+### Existing GitHub Workflows
+
+Existing workflows will evolve to implement the design principles. For example, if a new step is being added to an existing workflow, the step would be added via a Make command, which adheres to the principle that all logic is runnable on a developer machine.
+
+### New GitHub Workflows
+
+New GitHub workflows will adhere to the design principles.
+
+## Open Questions and Actions
```

Comment:
I missed the design meeting for this doc - Can you add answers to this questions (assuming they were answered in the design discussion) before checking this in?

Reply:
I updated the list to be a table with actions taken.